### PR TITLE
Add LBFGS optimizer of Paddle backend

### PR DESCRIPTION
**Pull request by xiaoguoguo626807**
add LBFGS optimizer for paddle backend, then more than 10 examples will be support. it is better to be review after 1076 PR merged

### CODE REVIEW

File: deepxde/optimizers/paddle/optimizers.py [hunk 1]
```diff
@@ -1,29 +1,83 @@
...
+        )
+    return lr_sch
 
 
 def is_external_optimizer(optimizer):
     return optimizer in ["L-BFGS", "L-BFGS-B"]
 
 
-def get(params, optimizer, learning_rate=None, decay=None):
+def get(params, optimizer, learning_rate=None, decay=None, weight_decay=0):
```
Comment (lululxvi) [inline deepxde/optimizers/paddle/optimizers.py:51]: Delete `weight_decay`.

File: examples/pinn_forward/Schrodinger.py [hunk 2]
```diff
@@ -0,0 +1,156 @@
+"""Backend supported: tensorflow.compat.v1, tensorflow, pytorch, paddle"""
```
Comment (lululxvi) [inline examples/pinn_forward/Schrodinger.py:1]: Add a separate PR for this new example.

File: deepxde/optimizers/paddle/optimizers.py (hunk 1 above)
Comment (xiaoguoguo626807) [reply deepxde/optimizers/paddle/optimizers.py]: done

File: examples/pinn_forward/Schrodinger.py (hunk 2 above)
Comment (xiaoguoguo626807) [reply examples/pinn_forward/Schrodinger.py]: done

Comment (xiaoguoguo626807) [reply examples/pinn_forward/Schrodinger.py]: https://github.com/lululxvi/deepxde/pull/1097

**Comment by lululxvi** [comment]
Please resolve the conflict.

File: deepxde/optimizers/config.py [hunk 3]
```diff
@@ -62,7 +62,7 @@ def set_LBFGS_options(
```
Comment (lululxvi) [inline deepxde/optimizers/config.py:62]: Update the docstring of `set_LBFGS_options`

File: deepxde/optimizers/paddle/optimizers.py [hunk 4]
```diff
@@ -1,6 +1,8 @@
 __all__ = ["get", "is_external_optimizer"]
 
 import paddle
+from paddle.incubate.optimizer import LBFGS
+from ..config import LBFGS_options
```
Comment (lululxvi) [inline deepxde/optimizers/paddle/optimizers.py:5]: Add an empty line:
```
from paddle.incubate.optimizer import LBFGS

from ..config import LBFGS_options
```

File: deepxde/optimizers/config.py (hunk 3 above)
Comment (xiaoguoguo626807) [reply deepxde/optimizers/config.py:63]: done

File: deepxde/optimizers/paddle/optimizers.py (hunk 4 above)
Comment (xiaoguoguo626807) [reply deepxde/optimizers/paddle/optimizers.py:6]: done

File: deepxde/optimizers/config.py (hunk 3 above)
Comment (lululxvi) [reply deepxde/optimizers/config.py:63]: Also update the `Args` part.

Comment (xiaoguoguo626807) [reply deepxde/optimizers/config.py:63]: done

File: deepxde/model.py [hunk 5]
```diff
@@ -472,11 +472,24 @@ def train_step(inputs, targets, auxiliary_vars):
             if self.lr_scheduler is not None:
                 self.lr_scheduler.step()
 
+        def train_step_lbfgs(inputs, targets, auxiliary_vars):
```
Comment (lululxvi) [inline deepxde/model.py:475]: Can this be merged with `train_step`? Pytorch supports this.

File: deepxde/model.py [hunk 6]
```diff
@@ -519,6 +532,8 @@ def _train_step(self, inputs, targets, auxiliary_vars):
         elif backend_name == "pytorch":
             # TODO: auxiliary_vars
             self.train_step(inputs, targets)
+        elif backend_name == "paddle":
```
Comment (lululxvi) [inline deepxde/model.py:535]: this is repeated with line 531

File: deepxde/model.py (hunk 5 above)
Comment (xiaoguoguo626807) [reply deepxde/model.py:475]: other optimizer in paddle use opt.step(), but lbfgs use opt.step(closure)

File: deepxde/model.py (hunk 6 above)
Comment (xiaoguoguo626807) [reply deepxde/model.py]: done

File: deepxde/model.py (hunk 5 above)
Comment (lululxvi) [reply deepxde/model.py:475]: Can other optimizers also use opt.step(closure)?

**Pull request merged**